### PR TITLE
Support some date/time operations.

### LINKED ISSUE #161: Document examples for temporal operations
The related PR is https://github.com/singularity-data/risingwave/pull/3472

### PR DESCRIPTION
**Pull request by bernscode**
Support some date/time operations.

## Info
- **Description**: 
Support some date/time operations:
date + int32
int32 + date
date - int32
date + time
time + date
time - time
time + interval
time - interval

- **Related code PR**: 
https://github.com/singularity-data/risingwave/pull/3472

- **Related doc issue**: 
Resolves https://github.com/singularity-data/risingwave-docs/issues/161

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked related doc issue to this PR in **Development**.
  - [x] I have accquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview and the updated parts look all good. <details><summary>How?</summary>image</details>
  - [x] (For version-specific PR) I have confirmed that this PR is for a released version. If the software version is not released, put it on hold.

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

**Review by emile-00 — APPROVED** [review]

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 1]
```diff
@@ -12,6 +12,13 @@ title: Date/time functions and operators
 | ----------- | ----------- | ----------- |
 | date + interval → timestamp | Add an interval to a date. | `date '2022-04-08' + interval '10 hour'` → `2022-04-08 10:00:00` |
 | date - interval → timestamp | Subtract an interval to a date. | `date '2022-04-08' - interval '10 hour'` → `2022-04-07 14:00:00` |
+| date + int → date | Add an int to a date. | `date '2022-06-23' + 4` → `2022-06-27` <br /> `4 + Date '2022-06-23'` → `2022-06-27` |
```
Comment (CharlieSYH) [inline docs/sql/functions-operators/sql-function-datetime.md]: Maybe we should avoid abbreviations in the description: 
"Add an integer to a date."

**Review by CharlieSYH — APPROVED** [review]

File: docs/sql/functions-operators/sql-function-datetime.md (hunk 1 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-datetime.md]: Good point! I've added and committed the change. Thanks!

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 2]
```diff
@@ -12,6 +12,13 @@ title: Date/time functions and operators
 | ----------- | ----------- | ----------- |
 | date + interval → timestamp | Add an interval to a date. | `date '2022-04-08' + interval '10 hour'` → `2022-04-08 10:00:00` |
 | date - interval → timestamp | Subtract an interval to a date. | `date '2022-04-08' - interval '10 hour'` → `2022-04-07 14:00:00` |
+| date + int → date | Add an integer to a date. | `date '2022-06-23' + 4` → `2022-06-27` <br /> `4 + Date '2022-06-23'` → `2022-06-27` |
+| date - int → date | Subtract an int from a date. | `date '2022-06-23' - 4` → `2022-06-19` |
```
Comment (hengm3467) [inline docs/sql/functions-operators/sql-function-datetime.md:16]: @bernscode Same comment as the one by Charlie. -> Subtract an integer from a date.

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 3]
```diff
@@ -12,6 +12,13 @@ title: Date/time functions and operators
 | ----------- | ----------- | ----------- |
 | date + interval → timestamp | Add an interval to a date. | `date '2022-04-08' + interval '10 hour'` → `2022-04-08 10:00:00` |
 | date - interval → timestamp | Subtract an interval to a date. | `date '2022-04-08' - interval '10 hour'` → `2022-04-07 14:00:00` |
+| date + int → date | Add an integer to a date. | `date '2022-06-23' + 4` → `2022-06-27` <br /> `4 + Date '2022-06-23'` → `2022-06-27` |
```
Comment (hengm3467) [inline docs/sql/functions-operators/sql-function-datetime.md]: This isn't clear. How about this: Add a number of days to a date.

File: docs/sql/functions-operators/sql-function-datetime.md (hunk 2 above)
Comment (hengm3467) [inline docs/sql/functions-operators/sql-function-datetime.md]: Same here. -- > Subtract a number of days from a date.

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 4]
```diff
@@ -12,6 +12,13 @@ title: Date/time functions and operators
...
+| date - int → date | Subtract an int from a date. | `date '2022-06-23' - 4` → `2022-06-19` |
+| date + time → timestamp | Add time to a date. | `date '2022-06-23' + time '19:24:00'` → `2022-06-23 19:24:00` <br /> `time '19:24:00' +  date '2022-06-23'` → `2022-06-23 19:24:00` |
```
Comment (hengm3467) [inline docs/sql/functions-operators/sql-function-datetime.md]: --> Add a time to a date.

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 5]
```diff
@@ -12,6 +12,13 @@ title: Date/time functions and operators
...
+| time - time → time | Subtract time from time. | `time '18:20:49' -  time '16:07:16'` → `02:13:33` |
```
Comment (hengm3467) [inline docs/sql/functions-operators/sql-function-datetime.md]: --> Subtract a time from a time;
(Or) Subtract times.

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 6]
```diff
@@ -12,6 +12,13 @@ title: Date/time functions and operators
...
+| time + interval → time | Add an interval to time. | `time '18:20:49' +  interval '1 hour'` → `19:20:49` |
```
Comment (hengm3467) [inline docs/sql/functions-operators/sql-function-datetime.md]: -> "Add an interval to a time."

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 7]
```diff
@@ -12,6 +12,13 @@ title: Date/time functions and operators
...
+| time - interval → time | Subtract an interval from time. | `time '18:20:49' -  interval '2 hours'` → `16:20:49` |
```
Comment (hengm3467) [inline docs/sql/functions-operators/sql-function-datetime.md]: -> "Subtract an interval from a time."

**Review by hengm3467 — CHANGES_REQUESTED** [review]

File: docs/sql/functions-operators/sql-function-datetime.md (hunk 3 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-datetime.md]: Good point! I agree, that does provide more clarity. I've updated this change. Thanks!

File: docs/sql/functions-operators/sql-function-datetime.md (hunk 2 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-datetime.md]: Updated change. Thanks!

File: docs/sql/functions-operators/sql-function-datetime.md (hunk 4 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-datetime.md]: Sounds good. Updated change. Thanks!

File: docs/sql/functions-operators/sql-function-datetime.md (hunk 5 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-datetime.md]: I think the former recommendation (Subtract a time from a time) works best here. Updated change. Thanks!

File: docs/sql/functions-operators/sql-function-datetime.md (hunk 6 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-datetime.md]: Updated change. Thanks!

**Review by hengm3467 — APPROVED** [review]
LGTM

**Pull request merged**